pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -5489,4 +5489,53 @@ protected CompletableFuture<AutoSubscriptionCreationOverride> internalGetAutoSub
                             return null;
                         }));
     }
+
+    protected CompletableFuture<MessageId> internalGetMessageIDByOffsetAsync(Long offset, boolean authoritative) {
```

Comment:
what happen if the index not enabled.

Reply:
The api will return a 404 error.

Reply:
It is better to check whether the index is enabled first, and then call this function to avoid unnecessary searches.